apps/src/lab2/views/components/Instructions/NavigationArea.tsx

```diff
@@ -162,27 +183,29 @@ const NavigationArea: React.FC<NavigationAreaProps> = ({
             />
           </div>
         )}
-        {canShowButton &&
-          (submittable ? (
-            <SubmitButton
-              levelId={id}
-              appName={appName}
-              disableEditRunForSubmission={disableEditRunForSubmission}
-              hasRun={hasRun}
-              hasEdited={hasEdited}
-              className={moduleStyles.buttonInstruction}
-            />
-          ) : (
-            <Button
-              id="instructions-continue-button"
-              size={'s'}
-              className={moduleStyles.buttonInstruction}
-              text={hasNextLevel ? commonI18n.continue() : commonI18n.finish()}
-              onClick={() => dispatch(continueOrFinishLesson())}
-              {...{type, color, iconRight}}
-            />
-          ))}
-        {showTts && feedbackMessage && !canShowButton && (
+        {submittable ? (
+          <SubmitButton
+            levelId={id}
+            appName={appName}
+            disableEditRunForSubmission={disableEditRunForSubmission}
+            requireRun={requireRun}
+            hasRun={hasRun}
+            hasEdited={hasEdited}
+            className={moduleStyles.buttonInstruction}
+          />
+        ) : (
+          <ContinueButton
+            disabled={!continueButtonIsEnabled}
+            type={type}
+            color={color}
+            iconRight={iconRight}
+            text={hasNextLevel ? commonI18n.continue() : commonI18n.finish()}
+            tooltipMessage={continueTooltipMessage}
+            hidden={hideContinueIfDisabled && !continueButtonIsEnabled}
```

Comment:
nit: this could just be a `hideIfDisabled` prop that receives `hideContinueIfDisabled`. Then inside `ContinueButton` you could just have
```
if (disabled && hideIfDisabled) {
  return null;
}
```

Reply:
Oh nice  - that makes sense. Will update!